bundles/org.openhab.core.io.rest.sse/src/main/java/org/openhab/core/io/rest/sse/internal/SseItemStatesEventBuilder.java

```diff
@@ -110,71 +108,76 @@ public SseItemStatesEventBuilder(final BundleContext bundleContext, final @Refer
-                                }
-                            }
+                }
+                if (pattern != null && !transformUsed && !optionUsed) {
+                    // if it's not a transformation pattern, then it must be a format string
+                    if (state instanceof QuantityType quantityState) {
+                        // sanity convert current state to the item state description unit in case it was
+                        // updated in the meantime. The item state is still in the "original" unit while the
+                        // state description will display the new unit:
+                        Unit<?> patternUnit = UnitUtils.parseUnit(pattern);
+                        if (patternUnit != null && !quantityState.getUnit().equals(patternUnit)) {
+                            quantityState = quantityState.toInvertibleUnit(patternUnit);
+                        }
 
-                            // The following exception handling has been added to work around a Java bug with formatting
-                            // numbers. See http://bugs.sun.com/view_bug.do?bug_id=6476425
-                            // This also handles IllegalFormatConversionException, which is a subclass of
-                            // IllegalArgument.
-                            try {
-                                displayState = state.format(pattern);
-                            } catch (IllegalArgumentException e) {
-                                logger.debug(
-                                        "Unable to format value '{}' of item {} with format '{}': {}, displaying raw state",
-                                        state, item.getName(), pattern, e.getMessage());
-                                displayState = state.toString();
-                            }
+                        if (quantityState != null) {
+                            state = quantityState;
                         }
+                    } else if (state instanceof DateTimeType type) {
+                        // Translate a DateTimeType state to the local time zone
+                        try {
+                            state = type.toLocaleZone();
+                        } catch (DateTimeException e) {
+                        }
+                    }
+
+                    // The following exception handling has been added to work around a Java bug with formatting
+                    // numbers. See http://bugs.sun.com/view_bug.do?bug_id=6476425
+                    // This also handles IllegalFormatConversionException, which is a subclass of
+                    // IllegalArgument.
+                    try {
+                        displayState = state.format(pattern);
+                    } catch (IllegalArgumentException e) {
+                        logger.debug(
+                                "Unable to format value '{}' of item {} with format '{}': {}, displaying raw state",
```

Comment:
```suggestion
                                "Unable to format value '{}' of item {} using format pattern '{}': {}, displaying raw state",
```

Reply:
Done.
I also updated accordingly the other very similar log message.